contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,376 @@
+            "Allowed slippage should not be over 10%"
+        );
+        require(
+            _uniswapV2CompatibleAddr != address(0),
+            "Uniswap compatible address should be non zero address"
+        );
+
+        RewardTokenConfig memory tokenConfig = RewardTokenConfig({
+            harvestRewardBps: _harvestRewardBps,
+            allowedSlippageBps: _allowedSlippageBps,
+            uniswapV2CompatibleAddr: _uniswapV2CompatibleAddr,
+            liquidationLimit: _liquidationLimit
+        });
+
+        address oldUniswapAddress = rewardTokenConfigs[_tokenAddress]
+            .uniswapV2CompatibleAddr;
+        rewardTokenConfigs[_tokenAddress] = tokenConfig;
+
+        IERC20 token = IERC20(_tokenAddress);
+
+        // if changing token swap provider cancel existing allowance
+        if (
+            oldUniswapAddress != address(0) &&
+            oldUniswapAddress != _uniswapV2CompatibleAddr
+        ) {
+            token.safeApprove(oldUniswapAddress, 0);
+        }
+
+        // Give Uniswap infinite approval when needed
+        if (_uniswapV2CompatibleAddr != address(0)) {
+            if (
+                token.allowance(address(this), _uniswapV2CompatibleAddr) <
+                type(uint256).max / 2
+            ) {
+                token.safeApprove(_uniswapV2CompatibleAddr, 0);
+                token.safeApprove(_uniswapV2CompatibleAddr, type(uint256).max);
+            }
+        }
+
+        emit RewardTokenConfigUpdated(
+            _tokenAddress,
+            _allowedSlippageBps,
+            _harvestRewardBps,
+            _uniswapV2CompatibleAddr,
+            _liquidationLimit
+        );
+    }
+
+    /***************************************
+                    Rewards
+    ****************************************/
+
+    /**
+     * @dev Transfer token to governor. Intended for recovering tokens stuck in
+     *      contract, i.e. mistaken sends.
+     * @param _asset Address for the asset
+     * @param _amount Amount of the asset to transfer
+     */
+    function transferToken(address _asset, uint256 _amount)
+        external
+        onlyGovernor
+    {
+        IERC20(_asset).safeTransfer(governor(), _amount);
+    }
+
+    /**
+     * @dev Collect reward tokens from all strategies
+     */
+    function harvest() external onlyGovernor nonReentrant {
+        _harvest();
+    }
+
+    /**
+     * @dev Swap all supported swap tokens for stablecoins via Uniswap.
+     */
+    function swap() external onlyGovernor nonReentrant {
+        _swap();
+    }
+
+    /*
+     * @dev Collect reward tokens from all strategies and swap for supported
+     *      stablecoin via Uniswap
+     */
+    function harvestAndSwap() external nonReentrant {
+        // TODO: add protection so that harvestAndSwap isn't called twice too closely together
+        _harvest();
+        _swap();
+        if (msg.sender != vaultAddress) {
+            IVault(vaultAddress).rebase();
```

Comment:
I may have missed some discussion on this, but do we need this rebase here? They seem to be two distinct actions. If the caller wants both of them they can call each separately, but it doesn't seem like we should mandate a rebase call here.

Reply:
I agree. I think we should keep the rebases out of the harvester.